diff --git a/src/plugins/file_system/s4u_FileSystem.cpp b/src/plugins/file_system/s4u_FileSystem.cpp
index 1a7ccccfbd..dc516b1b5e 100644
--- a/src/plugins/file_system/s4u_FileSystem.cpp
+++ b/src/plugins/file_system/s4u_FileSystem.cpp
@@ -121,7 +121,7 @@ void File::close()
   std::vector<int>* desc_table =
       Host::current()->extension<simgrid::s4u::FileDescriptorHostExt>()->file_descriptor_table.get();
   kernel::actor::simcall_answered([this, desc_table] { desc_table->push_back(this->desc_id); });
-  delete this;
+  // delete this;
 }
 
 void File::dump() const
diff --git a/src/smpi/bindings/smpi_pmpi_file.cpp b/src/smpi/bindings/smpi_pmpi_file.cpp
index e8dfd51da4..f3a6466cfb 100644
--- a/src/smpi/bindings/smpi_pmpi_file.cpp
+++ b/src/smpi/bindings/smpi_pmpi_file.cpp
@@ -43,13 +43,21 @@ int PMPI_File_open(MPI_Comm comm, const char *filename, int amode, MPI_Info info
   if (amode < 0)
     return MPI_ERR_AMODE;
   const SmpiBenchGuard suspend_bench;
+
+  aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
+  auto plop = "io-open " + std::string(filename);
+  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData(plop, amode));
   *fh =  new simgrid::smpi::File(comm, filename, amode, info);
+  TRACE_smpi_comm_out(rank_traced);
+
   if ((*fh)->size() != 0 && (amode & MPI_MODE_EXCL)){
     delete fh;
     return MPI_ERR_AMODE;
   }
   if(amode & MPI_MODE_APPEND)
     (*fh)->seek(0,MPI_SEEK_END);
+
+
   return MPI_SUCCESS;
 }
 
@@ -57,7 +65,11 @@ int PMPI_File_close(MPI_File *fh){
   CHECK_NULL(2, MPI_ERR_ARG, fh)
   CHECK_COLLECTIVE((*fh)->comm(), __func__)
   const SmpiBenchGuard suspend_bench;
+  aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
+  auto plop = "io-close " + std::string((*fh)->name());
+  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData(plop, 0));
   int ret = simgrid::smpi::File::close(fh);
+  TRACE_smpi_comm_out(rank_traced);
   *fh = MPI_FILE_NULL;
   return ret;
 }
@@ -65,7 +77,11 @@ int PMPI_File_close(MPI_File *fh){
 int PMPI_File_seek(MPI_File fh, MPI_Offset offset, int whence){
   CHECK_FILE(1, fh)
   const SmpiBenchGuard suspend_bench;
+  aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
+  auto plop = "io-seek " + std::string(fh->name());
+  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData(plop, 0));
   int ret = fh->seek(offset*fh->etype()->get_extent(),whence);
+  TRACE_smpi_comm_out(rank_traced);
   return ret;
 }
 
@@ -99,7 +115,8 @@ int PMPI_File_read(MPI_File fh, void *buf, int count,MPI_Datatype datatype, MPI_
   PASS_ZEROCOUNT(count)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
-  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData("IO - read", count * datatype->size()));
+  auto plop = "io-read " + std::string(fh->name());
+  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   int ret = simgrid::smpi::File::read(fh, buf, count, datatype, status);
   TRACE_smpi_comm_out(rank_traced);
   return ret;
@@ -111,8 +128,9 @@ int PMPI_File_read_shared(MPI_File fh, void *buf, int count,MPI_Datatype datatyp
   PASS_ZEROCOUNT(count)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
+  auto plop = "io-read_shared " + std::string(fh->name());
   TRACE_smpi_comm_in(rank_traced, __func__,
-                     new simgrid::instr::CpuTIData("IO - read_shared", count * datatype->size()));
+                     new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   int ret = simgrid::smpi::File::read_shared(fh, buf, count, datatype, status);
   TRACE_smpi_comm_out(rank_traced);
   return ret;
@@ -124,7 +142,8 @@ int PMPI_File_write(MPI_File fh, const void *buf, int count,MPI_Datatype datatyp
   PASS_ZEROCOUNT(count)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
-  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData("IO - write", count * datatype->size()));
+  auto plop = "io-write " + std::string(fh->name());
+  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   int ret = simgrid::smpi::File::write(fh, const_cast<void*>(buf), count, datatype, status);
   TRACE_smpi_comm_out(rank_traced);
   return ret;
@@ -136,8 +155,9 @@ int PMPI_File_write_shared(MPI_File fh, const void *buf, int count,MPI_Datatype
   PASS_ZEROCOUNT(count)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
+  auto plop = "io-write_shared " + std::string(fh->name());
   TRACE_smpi_comm_in(rank_traced, __func__,
-                     new simgrid::instr::CpuTIData("IO - write_shared", count * datatype->size()));
+                     new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   int ret = simgrid::smpi::File::write_shared(fh, buf, count, datatype, status);
   TRACE_smpi_comm_out(rank_traced);
   return ret;
@@ -149,7 +169,8 @@ int PMPI_File_read_all(MPI_File fh, void *buf, int count,MPI_Datatype datatype,
   CHECK_COLLECTIVE(fh->comm(), __func__)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
-  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData("IO - read_all", count * datatype->size()));
+  auto plop = "io-read_all " + std::string(fh->name());
+  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   int ret = fh->op_all<simgrid::smpi::File::read>(buf, count, datatype, status);
   TRACE_smpi_comm_out(rank_traced);
   return ret;
@@ -161,8 +182,9 @@ int PMPI_File_read_ordered(MPI_File fh, void *buf, int count,MPI_Datatype dataty
   CHECK_COLLECTIVE(fh->comm(), __func__)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
+  auto plop = "io-read_ordered " + std::string(fh->name());
   TRACE_smpi_comm_in(rank_traced, __func__,
-                     new simgrid::instr::CpuTIData("IO - read_ordered", count * datatype->size()));
+                     new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   int ret = simgrid::smpi::File::read_ordered(fh, buf, count, datatype, status);
   TRACE_smpi_comm_out(rank_traced);
   return ret;
@@ -174,7 +196,8 @@ int PMPI_File_write_all(MPI_File fh, const void *buf, int count,MPI_Datatype dat
   CHECK_COLLECTIVE(fh->comm(), __func__)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
-  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData("IO - write_all", count * datatype->size()));
+  auto plop = "io-write_all " + std::string(fh->name());
+  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   int ret = fh->op_all<simgrid::smpi::File::write>(const_cast<void*>(buf), count, datatype, status);
   TRACE_smpi_comm_out(rank_traced);
   return ret;
@@ -186,8 +209,9 @@ int PMPI_File_write_ordered(MPI_File fh, const void *buf, int count,MPI_Datatype
   CHECK_COLLECTIVE(fh->comm(), __func__)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
+  auto plop = "io-write_ordered " + std::string(fh->name());
   TRACE_smpi_comm_in(rank_traced, __func__,
-                     new simgrid::instr::CpuTIData("IO - write_ordered", count * datatype->size()));
+                     new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   int ret = simgrid::smpi::File::write_ordered(fh, buf, count, datatype, status);
   TRACE_smpi_comm_out(rank_traced);
   return ret;
@@ -199,7 +223,8 @@ int PMPI_File_read_at(MPI_File fh, MPI_Offset offset, void *buf, int count,MPI_D
   PASS_ZEROCOUNT(count)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
-  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData("IO - read", count * datatype->size()));
+  auto plop = "io-read_at " + std::string(fh->name()) + " " + std::to_string(offset);
+  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   MPI_Offset prev;
   fh->get_position(&prev);
   int ret = fh->seek(offset,MPI_SEEK_SET);
@@ -216,8 +241,9 @@ int PMPI_File_read_at_all(MPI_File fh, MPI_Offset offset, void *buf, int count,M
   CHECK_COLLECTIVE(fh->comm(), __func__)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
+  auto plop = "io-read_at_all " + std::string(fh->name()) + " " + std::to_string(offset);
   TRACE_smpi_comm_in(rank_traced, __func__,
-                     new simgrid::instr::CpuTIData("IO - read_at_all", count * datatype->size()));
+                     new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   MPI_Offset prev;
   fh->get_position(&prev);
   int ret = fh->seek(offset,MPI_SEEK_SET);
@@ -234,7 +260,8 @@ int PMPI_File_write_at(MPI_File fh, MPI_Offset offset, const void *buf, int coun
   PASS_ZEROCOUNT(count)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
-  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData("IO - write", count * datatype->size()));
+  auto plop = "io-write_at " + std::string(fh->name()) + " " + std::to_string(offset);
+  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   MPI_Offset prev;
   fh->get_position(&prev);
   int ret = fh->seek(offset,MPI_SEEK_SET);
@@ -251,8 +278,9 @@ int PMPI_File_write_at_all(MPI_File fh, MPI_Offset offset, const void *buf, int
   CHECK_COLLECTIVE(fh->comm(), __func__)
   const SmpiBenchGuard suspend_bench;
   aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
+  auto plop = "io-write_at_all " + std::string(fh->name()) + " " + std::to_string(offset);
   TRACE_smpi_comm_in(rank_traced, __func__,
-                     new simgrid::instr::CpuTIData("IO - write_at_all", count * datatype->size()));
+                     new simgrid::instr::CpuTIData(plop, count * datatype->size()));
   MPI_Offset prev;
   fh->get_position(&prev);
   int ret = fh->seek(offset,MPI_SEEK_SET);
@@ -266,7 +294,11 @@ int PMPI_File_write_at_all(MPI_File fh, MPI_Offset offset, const void *buf, int
 int PMPI_File_delete(const char *filename, MPI_Info info){
   CHECK_NULL(1, MPI_ERR_FILE, filename)
   const SmpiBenchGuard suspend_bench;
+  auto plop = "io-delete " + std::string(filename);
+  aid_t rank_traced = simgrid::s4u::this_actor::get_pid();
+  TRACE_smpi_comm_in(rank_traced, __func__, new simgrid::instr::CpuTIData(plop, 0));
   int ret = simgrid::smpi::File::del(filename, info);
+  TRACE_smpi_comm_out(rank_traced);
   return ret;
 }
 
diff --git a/src/smpi/include/smpi_file.hpp b/src/smpi/include/smpi_file.hpp
index 60ab2fde22..c74e3e0855 100644
--- a/src/smpi/include/smpi_file.hpp
+++ b/src/smpi/include/smpi_file.hpp
@@ -44,7 +44,8 @@ class File : public F2C{
   int flags() const;
   MPI_Datatype etype() const;
   MPI_Comm comm() const;
-  std::string name() const override {return file_ ? std::string("MPI_File: ")+ std::string(file_->get_path()): std::string("MPI_File");}
+  // std::string name() const override {return file_ ? std::string("MPI_File: ")+ std::string(file_->get_path()): std::string("MPI_File");}
+  std::string name() const override {return file_ ? std::string(file_->get_path()): std::string("MPI_File");}
 
   int sync();
   int seek(MPI_Offset offset, int whence);
@@ -142,7 +143,7 @@ int File::op_all(void* buf, int count, const Datatype* datatype, MPI_Status* sta
       //we want to send only useful data, so let's pretend we pack it
       send_sizes[i]=send_sizes[i]/datatype->get_extent()*datatype->size();
       // store min and max offset to actually read
-  
+
       min_offset = std::min(min_offset, min_offsets[i]);
       total_sent += send_sizes[i];
       XBT_CDEBUG(smpi_io, "will have to send %d bytes to %d", send_sizes[i], i);
